src/main/kotlin/org/jetbrains/research/ml/codetracker/reporting/GitHubErrorReporter.kt

```diff
@@ -0,0 +1,95 @@
+import com.intellij.openapi.project.Project
+import com.intellij.util.Consumer
+import java.awt.Component
+
+class GitHubErrorReporter : ErrorReportSubmitter() {
+    override fun submit(
+        events: Array<IdeaLoggingEvent>, additionalInfo: String?, parentComponent: Component,
+        consumer: Consumer<SubmittedReportInfo>
+    ): Boolean {
+        val errorReportInformation = ErrorInformation(
+            IdeaLogger.ourLastActionId,
+            ApplicationInfo.getInstance() as ApplicationInfoEx,
+            ApplicationNamesInfo.getInstance(),
+            events[0].throwable
+        )
+        return doSubmit(
+            events[0],
+            parentComponent,
+            consumer,
+            errorReportInformation
+        )
+    }
+
+    override fun getReportActionText(): String {
+        return "Report to Codetracker Github Issue Tracker"
+    }
+
+    /**
+     * Provides functionality to show an error report message to the user that gives a click-able link to the created issue.
+     */
+    internal class CallbackWithNotification(
+        private val originalConsumer: Consumer<SubmittedReportInfo>,
+        private val project: Project?
+    ) :
+        Consumer<SubmittedReportInfo> {
+        override fun consume(reportInfo: SubmittedReportInfo) {
+            originalConsumer.consume(reportInfo)
+            val notificationType =
+                if (reportInfo.status == SubmittedReportInfo.SubmissionStatus.FAILED) NotificationType.ERROR else NotificationType.INFORMATION
+            ReportMessages.GROUP.createNotification(
+                ReportMessages.ERROR_REPORT,
+                reportInfo.linkText,
+                notificationType,
+                NotificationListener.URL_OPENING_LISTENER
+            ).setImportant(false).notify(project)
+        }
+
+    }
+
+    companion object {
+        private fun doSubmit(
+            event: IdeaLoggingEvent,
+            parentComponent: Component,
+            callback: Consumer<SubmittedReportInfo>,
+            errorInformation: ErrorInformation
+        ): Boolean {
+            if (event is IdeaReportingEvent) {
+                event.plugin?.let {
+                    errorInformation.setUserInformation(UserInformationType.PLUGIN_NAME, it.name)
+                    errorInformation.setUserInformation(UserInformationType.PLUGIN_VERSION, it.version)
+                }
+            }
+            val dataContext = DataManager.getInstance().getDataContext(parentComponent)
+            val project = CommonDataKeys.PROJECT.getData(dataContext)
+            val task = ErrorReportTask(
+                project,
+                "Submitting error report",
+                true,
+                errorInformation,
+                CallbackWithNotification(callback, project)
+            )
+            project?.let { ProgressManager.getInstance().run(task) } ?: run {
+                task.run(EmptyProgressIndicator())
+            }
```

Comment:
I think in case of small blocks it's preferable to use 'let' as you did. See https://stackoverflow.com/questions/45877140/kotlin-and-idiomatic-way-to-write-if-not-null-else-based-around-mutable-v